### ISSUE TITLE
Flatpak build

### ISSUE THREAD

**Issue by ja-cop** [virtual-puppet-project/vpuppr#107]
First of all, thanks a bunch for this awesome software! I got it working on Void Linux by compiling Python 3.8 myself (Void comes with 3.10 in the package repository), per issue #93. I used [VRM_Addon_for_Blender](https://github.com/saturday06/VRM_Addon_for_Blender) to configure and export a VRM directly from Blender - no intermediate software was required. Nice!

Perhaps releasing the app as a [Flatpak](https://flatpak.org/index.html) on Flathub ([App Submission docs](https://github.com/flathub/flathub/wiki/App-Submission)) would be a good idea? The appropriate Python and libc etc. dependencies could be included or pulled in from a runtime (AFAIK), and maybe the Python packages in the venv could be pre-installed, too. Additional niceties provided by this approach includes easy updates for users, and automatic install/uninstall of a .desktop entry for launching the application.

I might have a go later at building a Flatpak and see what it takes, but in any case, this issue could serve as the place to discuss Flatpak.

**Comment by you-win** [comment]
Glad to see my efforts for Linux compatibility weren't in vain and thanks for trying out the app!

**tl;dr I'll look into it, and I'm open to pointers/suggestions/examples**

I'll have to take a look at the submission docs for Flathub. I'm not very familiar with packaging applications on Linux besides just dropping stuff into `/opt` 🙃.

Regarding bundling in dependencies, I'm also not sure what the best approach. I considered bundling a Python interpreter + the various python packages needed on Linux, but I thought it would be best to let the system's `pip` installation sort out the correct dependencies.

I'm wondering if a CICD pipeline for handling all the various packaging things (Flathub, regular `.zip` releases, etc) is feasible, since currently each new release is cut manually be me.

**Comment by mostlyfireproof** [comment]
I'm also interested in this! This is the only 3d face tracking program I've found that works natively on Linux, so I'd love an easy install so I don't need to screw with different Python versions.

**Comment by Doomsdayrs** [comment]
Good idea to start here https://docs.flatpak.org/en/latest/

**Comment by Doomsdayrs** [comment]
And python https://docs.flatpak.org/en/latest/python.html?highlight=python#python

**Comment by ByteDream** [comment]
A flatpak was recently created and seems to work flawless (and even fixes bugs on some machines which occurred  when using vpuppr from the GH releases natively).
It is available on [flathub](https://flathub.org/apps/details/com.github.virtual_puppet_project.vpuppr).

**Comment by orowith2os** [comment]
The flatpak is up, and seems to be working without issues. Will be working on an appimage if possible, just to be sure everybody can have something that Just Works for their system. Snaps aren't in my list of goals yet, at least until it's open sourced.
